std/assembly/typedarray.ts

```diff
@@ -1260,3 +1304,34 @@ function REVERSE<TArray extends ArrayBufferView, T>(array: TArray): TArray {
   }
   return array;
 }
+
+// @ts-ignore: decorator
+@inline
+function WRAP<TArray extends ArrayBufferView, T>(buffer: ArrayBuffer, byteOffset: i32 = 0, length: i32 = -1): TArray {
+  if (byteOffset < 0 || byteOffset >= buffer.byteLength) {
+    throw new RangeError("Start offset out of bound");
+  }
+  var byteLength: i32;
+  if (length < 0) {
+    if (length == -1) {
+      let size = sizeof<T>();
+      if (buffer.byteLength % size == 0) {
+        byteLength = buffer.byteLength;
+      } else {
+        throw new RangeError("Byte length is not a multiple of " + size.toString());
```

Comment:
Maybe just `E_INVALIDLENGTH` as well. Unless someone finds the message particularly important, then it might become a something new like `E_LENGTHMISMATCH`.

Reply:
Fixed